azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventData.java

```diff
@@ -322,18 +323,22 @@ private void writeObject(ObjectOutputStream out) throws IOException
 	{
 		out.defaultWriteObject();
 		
-		out.writeInt(this.bodyData.getLength());
-		out.write(this.bodyData.getArray(), this.bodyData.getArrayOffset(), this.bodyData.getLength());
+                out.writeInt(this.bodyData == null ? BODY_DATA_NULL : this.bodyData.getLength());
+                if (this.bodyData != null)
+                    out.write(this.bodyData.getArray(), this.bodyData.getArrayOffset(), this.bodyData.getLength());
 	}
 	
 	private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException
 	{
 		in.defaultReadObject();
 		
 		final int length = in.readInt();
-		final byte[] data = new byte[length];
-		in.read(data, 0, length);
-		this.bodyData = new Binary(data, 0, length);
+		if (length != BODY_DATA_NULL) {
+                    
+                    final byte[] data = new byte[length];
+                    in.readFully(data, 0, length);
```

Comment:
>in.readFully( [](start = 20, length = 13)

Do we actually validate body data in our stress tests? We have CRC check for .NET cases, we should have the same tests ported to Java.